svn/trunk/SlickEngagement_Plugin.php

```diff
@@ -293,135 +262,199 @@ public function removeSemicolons($value)
+            
+            echo "\n<script>\n";
+            echo "/* Slickstream CLS Insertion */\n";
+            echo '"use strict";(async(e,t)=>{const n=e?JSON.parse(e):null;const r=t?JSON.parse(t):null;if(n||r){const e=async()=>{if(document.body){if(n){o(n.selector,n.position||"after selector","slick-film-strip",n.minHeight||72)}if(r){r.forEach((e=>{if(e.selector){o(e.selector,e.position||"after selector","slick-inline-search-panel",e.minHeight||350,e.id)}}))}return}window.requestAnimationFrame(e)};window.requestAnimationFrame(e)}const c=async(e,t)=>{const n=Date.now();while(true){const r=document.querySelector(e);if(r){return r}const c=Date.now();if(c-n>=t){throw new Error("Timeout")}await i(200)}};const i=async e=>new Promise((t=>{setTimeout(t,e)}));const o=async(e,t,n,r,i)=>{try{const o=await c(e,5e3);const s=i?document.querySelector(`.${n}[data-config="${i}"]`):document.querySelector(`.${n}`);if(o&&!s){const e=document.createElement("div");e.style.minHeight=r+"px";e.classList.add(n);if(i){e.dataset.config=i}switch(t){case"after selector":o.insertAdjacentElement("afterend",e);break;case"before selector":o.insertAdjacentElement("beforebegin",e);break;case"first child of selector":o.insertAdjacentElement("afterbegin",e);break;case"last child of selector":o.insertAdjacentElement("beforeend",e);break}return e}}catch(t){console.log("plugin","error",`Failed to inject ${n} for selector ${e}`)}return false}})' . "\n";
+            echo "('" . addslashes($filmstrip_str) . "','" . addslashes($dcm_str) . "');" . "\n";
+            echo "\n</script>\n";
+        }
+    }
+
+    //Returns a name for the "page boot data" transient 
+    private function getTransientName()
+    {
+        DEFINE('PAGE_BOOT_DATA_TRANSIENT_PREFIX', 'slick_page_boot_');
+        $normalized_url = $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
+        return PAGE_BOOT_DATA_TRANSIENT_PREFIX . md5($normalized_url);
+    }
+
+    private function echoSlickstreamComment($comment)
+    {
+        echo "\n<!-- [Slickstream] " . $comment . " -->\n";
+    }
+
+    private function getCurrentTimestampByTimeZone($tz_name)
+    {
+        $timestamp = time();
+        $dt = new DateTime('now', new DateTimeZone($tz_name));
+        $dt->setTimestamp($timestamp);
+        return $dt->format('Y-m-d H:i:s');
+    }
+
+    private function echoPageBootData() 
+    {
+        $siteCode = substr(trim($this->getOption('SiteCode')), 0, 9);
+
+        if (!$siteCode) {
+            $this->echoSlickstreamComment("ERROR: Site Code missing from Plugin Settings; Slickstream services are disabled");
+            return;
+        }
+
+        global $wp;
+        
+        $transient_name = $this->getTransientName(); //Name for WP Transient Cache API Key
+
+        // Check for existing data in transient cache. If none, then fetch data from server.
+        if (false === ($boot_data_obj = get_transient($transient_name))) {
```

Comment:
what do you think of adding a query param, so we have option to not read from cache and fetch from server to update cached data to latest instantly?